src/events/http2.js

```diff
@@ -0,0 +1,169 @@
+    URL_BLACKLIST,
+    generateEpsagonTraceId,
+    updateAPIGateway,
+} = require('../helpers/http');
+const tryRequire = require('../try_require');
+
+const http2 = tryRequire('http2');
+
+
+/**
+ * http2 module adds in the request and response headers also extra fields with a ':' prefix.
+ * For example :path, :method, etc. When we want to record just the headers, we want to clear
+ * out these fields - using this function.
+ * @param {object} headers data
+ * @returns {object} only real headers without ':'
+ */
+function extractHeaders(headers) {
+    return Object.entries(headers) // Iterate over key-value pairs
+        .filter(header => !header[0].startsWith(':')) // Filter out keys that start with ':'
+        .reduce((obj, header) => { // Rebuild key-value into object using reduce
+            const [key, value] = header;
+            obj[key] = value; // eslint-disable-line no-param-reassign
+            return obj;
+        }, {});
+}
+
+/**
+ * Wraps the http2's module request function with tracing
+ * @param {Function} wrappedFunction The http2's request module
+ * @param {string} authority hostname
+ * @returns {Function} The wrapped function
+ */
+function httpWrapper(wrappedFunction, authority) {
+    return function internalHttpWrapper(headers, options) {
+        let clientRequest = null;
+        try {
+            const { hostname } = urlLib.parse(authority);
+
+            const reqHeaders = extractHeaders(headers);
+
+            if (isBlacklistURL(hostname, URL_BLACKLIST, headers[':path']) || isURLIgnoredByUser(hostname)) {
+                utils.debugLog(`filtered blacklist hostname ${hostname}`);
+                return wrappedFunction.apply(this, [headers, options]);
+            }
+            if (isBlacklistHeader(reqHeaders, USER_AGENTS_BLACKLIST)) {
+                utils.debugLog('filtered blacklist headers');
+                return wrappedFunction.apply(this, [headers, options]);
+            }
+
+            // Inject header to support tracing over HTTP requests to opentracing monitored code
+            const epsagonTraceId = generateEpsagonTraceId();
+            headers['epsagon-trace-id'] = epsagonTraceId; // eslint-disable-line no-param-reassign
+
+            const { slsEvent: httpEvent, startTime, resource } = eventInterface.initializeEvent(
+                'http',
+                hostname,
+                headers[':method'],
+                'http'
+            );
+
+            eventInterface.addToMetadata(httpEvent,
+                {
+                    http_trace_id: epsagonTraceId,
+                }, {
+                    path: headers[':path'],
+                    request_headers: reqHeaders,
+                });
+
+            try {
+                clientRequest = wrappedFunction.apply(this, [headers, options]);
+            } catch (err) {
+                eventInterface.setException(httpEvent, err);
+                tracer.addEvent(httpEvent);
+                throw err;
```

Comment:
This will be caught by the external try/catch and not be propagated to the user.